### PR TITLE
VSCode Extension

### PR DESCRIPTION
**Pull request by waydegg**
Adds a chat-like VSCode Extension with a simple language server interfacing with Mentat.

### CODE REVIEW

File: mentat/language_server.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+                method="mentat/inputRequest",
+                data=input_request_message,
+            )
+            ls.send_notification("mentat/inputRequest", ls_message.model_dump(mode="json"))
+
+    ls.handle_session_stream_task = asyncio.create_task(handle_session_stream())
+    ls.handle_input_requests_task = asyncio.create_task(handle_input_requests())
+
+    print("Waiting for input request")
+    input_request_message = await ls.session.stream.recv("input_request")
```
Comment (PCSwingle) [inline mentat/language_server.py:189]: What is this for?

File: mentat/language_server.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+
+@server.feature("mentat/createSession")
+async def create_session(ls: LanguageServer, params: lsp.InitializeParams):
+    assert ls.cwd is not None
+
+    ls.session = Session(cwd=ls.cwd)
+    ls.session.start()
+
+    async def handle_session_stream():
```
Comment (PCSwingle) [inline mentat/language_server.py:163]: Why do we need 2 separate listeners? Maybe I'm misunderstanding how this is supposed to work but I would assume that the point of this LanguageServer class is to forward stream messages to the vscode extension and forward responses from the vscode extension back to the session stream

File: mentat/language_server.py [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+
+class Message(NamedTuple):
+    data: Any
+
+
+@server.feature("mentat/clientMessage")
+async def handle_client_message(ls: LanguageServer, message: Any):
+    if ls.session is None:
+        return
+    if ls.session_input_request_id is None:
```
Comment (PCSwingle) [inline mentat/language_server.py:217]: This shouldn't care about the session input request id, right? Shouldn't it just forward all received messages from the vscode extension to the stream?

**Comment by PCSwingle** [comment]
There are a whole bunch of changes to the python code where your formatter is changing the default format (and presumably causing it to fail black and isort tests); it would make it a lot easier to review if you could reformat with black and isort so that those changes don't show up in the git diff.

**Comment by PCSwingle** [comment]
Just tested it out; looks great so far! I'm ok with releasing a very basic version, but a few things I think we need to fix before we release:

1. The first time I ran it, I got this error in the bottom right: 
```
Activating extension 'AbanteAI.mentat' failed: Cannot find module '/home/pcs/mentat/mentat-vscode/build/extension.js'
Require stack:
- /usr/lib/code/out/vs/loader.js
- /usr/lib/code/out/bootstrap-amd.js
- /usr/lib/code/out/bootstrap-fork.js
- .
```
After I restarted and reran it it all worked though? So I'm not sure what's going on here.

2. Colors in output

3. Output formatting. Whenever we output code, we need to output it in a code block. Right now indentation is completely ignore and multiple spaces are collapsed into 1 (since that's what HTML does); to fix this we need to put it in a pre block instead of a span or p or whatever we're using right now. Long term though I think it would look nicer in a code block.

4(?). This could probably wait until after release, but it would be really nice to change the UI for things like Yes no interaction to maybe buttons. 

5. Some output randomly has a black background? The context display for example. I haven't looked too closely at the JS code yet so I don't know what's going on, but it would be good if it was all consistent.

File: mentat-vscode/src/lib/WebviewProvider.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
+import emitter from "utils/emitter"
+import { LanguageServerMessage } from "types"
+import * as vscode from "vscode"
+import { Uri, Webview } from "vscode"
+
+import { LanguageClient } from "vscode-languageclient/node"
+
+/**
```
[the remark this answers is not in the document]
Comment (waydegg) [reply mentat-vscode/src/lib/WebviewProvider.ts:8]: This function is from [this](https://github.com/microsoft/vscode-webview-ui-toolkit-samples/blob/main/frameworks/component-gallery-react/src/utilities/getNonce.ts) extension template from Microsoft.

File: mentat-vscode/src/utils/setup.ts [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+  const serverOptions = await createMentatSocket({
+    host: languageServerHost,
+    port: languageServerPort,
+  })
+
+  return serverOptions
+}
+
+export { installMentat, getLanguageServerOptions }
```
[the remark this answers is not in the document]
Comment (waydegg) [reply mentat-vscode/src/utils/setup.ts:151]: Correct, need to update this

File: mentat-vscode/scripts/build.mjs [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  const now = new Date();
+  const year = now.getFullYear();
+  const month = (now.getMonth() + 1).toString().padStart(2, "0");
+  const day = now.getDate().toString().padStart(2, "0");
+  const hours = now.getHours().toString().padStart(2, "0");
+  const minutes = now.getMinutes().toString().padStart(2, "0");
+  const seconds = now.getSeconds().toString().padStart(2, "0");
+  const formattedDate = `${year}-${month}-${day} ${hours}:${minutes}:${seconds}`;
+  return formattedDate;
+}
```
[the remark this answers is not in the document]
Comment (waydegg) [reply mentat-vscode/scripts/build.mjs:15]: Maybe there's an external package for this? I'm not sure though.

**Review by granawkins — COMMENTED** [review]
LGTM so far

**Pull request merged**